software-mansion/react-native-reanimated
# Enable build from source on client side

## Description

This PR introduces mechanism to build aar for reanimated, from source on a client side, in case appropriate version is not provided initially with the package.

## Changes

* Merged `android-npm/build.gradle` with `android/build.gradle`
* Adjusted paths in `android/CMakeLists.txt`
* Updated `createNPMPackage` script

  Android source code is now provided with the package

## Before

For given react-native minor version, if aar built for equal of lower was not provided, the build would fail.

## After

If aar built for equal or lower react-native minor version is not provided, client attempts to build it from the provided source code. 

## Test code and steps to reproduce

Changes were tested on [reanimated react native version tester](https://github.com/piaskowyk/reanimated-rn-version-tester).

1. Build package for some react-native version
2. See that projects for different RN versions compile 



## Checklist
